pkg/operation/botanist/component/extensions/worker/worker_test.go

```diff
@@ -538,6 +540,67 @@ var _ = Describe("Worker", func() {
 		})
 	})
 
+	Describe("#WaitUntilWorkerStatusMachineDeploymentsUpdate", func() {
+		It("should return error when no resources are found", func() {
+			Expect(defaultDepWaiter.WaitUntilWorkerStatusMachineDeploymentsUpdate(ctx)).To(HaveOccurred())
+		})
+
+		It("should return error when status.machineDeploymentsLastUpdateTime is not updated", func() {
+			obj := w.DeepCopy()
+			Expect(c.Create(ctx, obj)).To(Succeed(), "creating worker succeeds")
+
+			Expect(defaultDepWaiter.WaitUntilWorkerStatusMachineDeploymentsUpdate(ctx)).To(HaveOccurred(), "worker status is not updated")
+		})
+
+		It("should return error if we haven't observed the latest timestamp annotation", func() {
```

Comment:
Why do we want to test the timestamp annotation? Shouldn't we rather test that it fails if the `machineDeploymentsLastUpdateTime` remains the same?

Reply:
Added unit test for this in [cd22f4a](https://github.com/gardener/gardener/pull/7774/commits/cd22f4aa21ae5691caa5952884c1599029f0ac14). I kept the timestamp one because that check is part of `WaitUntilObjectReadyWithHealthFunction` which is called inside ` WaitUntilWorkerStatusMachineDeploymentsUpdate` but, now I have removed it.